source/model/table.py

```diff
@@ -0,0 +1,30 @@
+import json
+import os
+from model.index import *
+from commands.keys import *
+from output.exceptions import *
+
+
+class TableMetadata:
+    def __init__(self, table_map, path):
+        self.table_map = table_map
+        self.path = os.path.join(path, self.table_map[keys.NAME])
+        self.__validate__(self)
+        self.serialize()
+
+    def serialize(self):
+        os.makedirs(self.path, exist_ok=True)
+        with open(os.path.join(self.path, "{}.json".format(self.table_map[keys.NAME])), 'w') as file:
+            json.dump(self.table_map, file)
+        self.__create_indices__(self)
+
+    @staticmethod
+    def __create_indices__(self):
+        for index in self.table_map[keys.IDX_KEY]:
+            index = Index(index, self.table_map, self.path)
```

Comment:
Index should have a Table object not the table map.
You didn’t follow encapsulation concept